cli/cmd/dashboard.go

```diff
@@ -23,50 +23,40 @@ var dashboardCmd = &cobra.Command{
 			return fmt.Errorf("port must be positive, was %d", proxyPort)
 		}
 
-		kubectl, err := k8s.NewKubectl(shell.NewUnixShell())
+		api, err := k8s.NewK8sAPI(shell.NewUnixShell().HomeDir(), kubeconfigPath)
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "Failed to start kubectl: %s", err)
 			os.Exit(1)
 		}
 
-		asyncProcessErr := make(chan error, 1)
-
-		err = kubectl.StartProxy(asyncProcessErr, proxyPort)
-
+		url, err := api.ProxyUrlFor(controlPlaneNamespace, "/services/web:http/proxy/", proxyPort)
 		if err != nil {
-			fmt.Fprintf(os.Stderr, "Failed to start kubectl proxy: %s", err)
-			os.Exit(1)
-		}
-
-		url, err := kubectl.UrlFor(controlPlaneNamespace, "/services/web:http/proxy/")
 
-		if err != nil {
 			fmt.Fprintf(os.Stderr, "Failed to generate URL for dashboard: %s", err)
 			os.Exit(1)
 		}
 
 		fmt.Printf("Opening [%s] in the default browser\n", url)
 		err = browser.OpenURL(url.String())
-
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "Failed to open URL %s in the default browser: %s", url, err)
 			os.Exit(1)
 		}
 
-		select {
-		case err = <-asyncProcessErr:
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "Error starting proxy via kubectl: %s", err)
-				os.Exit(1)
-			}
+		// blocks until killed
+		err = api.StartProxy(proxyPort)
```

Comment:
is it possible to use a port of `0` so it binds an ephemeral port? If it doesn't return, then I'm guessing it's not practical to get the port that way...

If not, can we manually try to choose an ephemeral port until there are no port conflicts? (i.e. by handling the port-conflict error and retrying on a new port)